src/pages/Fundraising/Fundraising.tsx

```diff
@@ -157,6 +202,14 @@ export const FundraisingPage: React.FC = () => {
               <li>{`USDC Contributed: ${utils.commify(utils.formatUnits(rewards.price, 6))}`}</li>
               <li>{`SHER Reward: ${utils.commify(utils.formatUnits(rewards.sherAmount, 18))} tokens`}</li>
             </ul>
+            <AllowanceGate
+              allowance={allowance}
```

Comment:
I know it's a PITA the way `AllowanceGate` behaves at the moment. I was looking to have it take care of getting the current allowance, but everytime I got into some sort of infinite loops in the interaction with wagmi, having `useERC20` hook being used in nested components, so I resorted to just passing the allowance as a prop.

Reply:
haha yeh, there's a bit of repeated logic around allowances in both here and `Protocol.tsx` but we can refactor that component later on.